### DISCUSSION TITLE
Power USB Code

### DISCUSSION THREAD

**Discussion by ar8001** [gopro/labs#832] [Q&A]
I think i'm to stupid to understand the power usb start code ...

My wish is:
if the usb-power is connected to the mini11 that the gopro start in timewarp / timelapse mode and record the timewarp or timelapse video until i remove the power it stops ... 
the !S command doesnt work for me:

!SmTp.10r4XfL0

**Comment by dnewman-gpsw** [comment]
```!S``` this the start command, nothing else.  You do have your command backward, and you start capture then change the settings, which is not supported.  Also this is not going to make the camera start on power.  

This is a very complex example of what you can do: https://www.youtube.com/shorts/jH5Xr86_Qrs

You need to set ```!MWAKE=2```  <- to boot on USB power.
You also need a boot command, to tell the command what to do.  

Start with simple with Blockly to script the camera https://gopro.github.io/labs/build/

This is untested (just as an example)
![image](https://github.com/gopro/labs/assets/10856084/00c5d4c3-a017-4667-8ab9-fd6ded0ed53a)

![TimeWarp on Power-4_1_2024, 09_37_20](https://github.com/gopro/labs/assets/10856084/03228d57-3bd3-45c8-9381-603f2b1fd14c)

**Comment by ar8001** [comment]
thanks for your fast answer...
but it doesnt'w ork here. 

i have the latest firmware and labs update on the 11 mini.

if i use the code, the mini display turns to black and beeb 3 times all 15 seconds ...   thats all
if i remove usb power ... still 3 times beeb all 15 seconds

**Reply by dnewman-gpsw** [reply]
I just tested it, it works fine.  I figure you are missing a step, so here it is combined into one QR Code.

 
![TimeWarp on Power-4_1_2024, 10_46_20](https://github.com/gopro/labs/assets/10856084/c173c1ed-7813-45b9-a173-01374c1e4d7c)

**Reply by ar8001** [reply]
Thanks a lot!
I'll check later

**Reply by ar8001** [reply]
if i like to change the power on start to timelapse photo i only hav to change the TW Command ?

**Reply by dnewman-gpsw** [reply]
Mostly, mTW is mode Time Warp.  Timelapse is mTP.  With this disclaimer:

Time/Night-Lapse modes: The camera defaults to having presets for only one time-lapse mode and one night-lapse mode, yet both modes have two configurations (video or photo.) For all QR Code mode commands to work, a preset is needed for every mode you wish to use. e.g. If you only have a preset for night-lapse video (NLV), the QR command for NLV will work, but will not for night-lapse photo (NLP). To solve this, create a new preset for the timelapse more you need, then both video and photo time-lapse commands will work.
